plugins/carbon_black_cloud/icon_carbon_black_cloud/connection/connection.py

```diff
@@ -146,3 +152,72 @@ def test(self):
                 data=str(error),
             )
         return {"success": True}
+
+    def test_task(self):
+        self.logger.info("Running a connection test to Carbon Black Cloud")
+
+        alerts_endpoint = self.base_url + f"/api/alerts/v7/orgs/{self.org_key}/alerts/_search"
+        observations_endpoint = self.base_url + f"/api/investigate/v2/orgs/{self.org_key}/observations/search_jobs"
+
+        current_time = datetime.now(timezone.utc)
+        now = current_time.strftime(TIME_FORMAT)
+        minus_five_mins = (current_time - timedelta(minutes=5)).strftime(TIME_FORMAT)
+
+        fd = {
+            alerts_endpoint: {
+                "time_range": {"start": minus_five_mins, "end": now},
+                "criteria": {},
+                "start": "1",
+                "rows": str(1),  # max number of results that can be returned
+                "sort": [{"field": ALERT_TIME_FIELD, "order": "ASC"}],
+            },
+            observations_endpoint: {
+                "rows": 1,
+                "start": 0,
+                "fields": ["*"],
+                "criteria": {"observation_type": OBSERVATION_TYPES},
+                "sort": [{"field": OBSERVATION_TIME_FIELD, "order": "asc"}],
+                "time_range": {"start": minus_five_mins, "end": now},
+            },
+        }
+
+        return_msg = "Task connection test to Carbon Black Cloud successful"
+        return_msg_error = "Task connection test to Carbon Black Cloud failed"
+        self.logger.info("Testing get alerts")
+
+        failed = ""
+        failed_endpoint = []
+
+        for key, value in fd.items():
+            try:
+                # Get a 200 from the search jobs endpoint
+                self.request_api(key, value)
+
+            except HTTPErrorException as error:
+                # If we get a similar error for the second URL
+                if error.cause in failed:
+                    failed_endpoint.append(key)
+                else:
+                    return_msg_error += f"\n{error.cause}\n{error.assistance}"
+                    failed_endpoint.append(key)
+                    failed += return_msg_error
+
+            except RateLimitException as error:
+                return_msg_error += "\nToo many requests."
+                return_msg_error += "\nPlease slow down"
+                raise ConnectionTestException(cause=error.cause, assistance=error.assistance, data=return_msg_error)
+
+            except PluginException as error:
+                return_msg_error += "\nRetry on 503 failed."
+                raise ConnectionTestException(cause=error.cause, assistance=error.assistance, data=return_msg_error)
+
+        if failed:
+            if len(failed_endpoint) > 1:
+                failed += "\nThe endpoint for alerts and observations failed."
+            else:
+                for item in failed_endpoint:
+                    failed += f"\nThe endpoint for {item.rstrip('_')} failed"
```

Comment:
I think this is going to still include the whole URL? We might need to update `fd` definition so that the key is `alerts` and `observations` and then pass the endpoint/params as the values in the loop?

Also think we should include cause/assistance in the raise incase we decide to implement this further down the line / or if use it now

Reply:
Not sure what to include in the cause and assist now since we add the errors cause and assist in the output string for data.